src/libsystemd/sd-device/sd-device.c

```diff
@@ -345,9 +345,11 @@ _public_ int sd_device_new_from_ifname(sd_device **ret, const char *ifname) {
         assert_return(ret, -EINVAL);
         assert_return(ifname, -EINVAL);
 
-        r = device_new_from_main_ifname(ret, ifname);
-        if (r >= 0)
-                return r;
+        if (ifname_valid(ifname)) {
+                r = device_new_from_main_ifname(ret, ifname);
+                if (r >= 0)
+                        return r;
+        }
```

Comment:
(maybe the lesson to learn here, is to switch device_new_from_main_ifname() to use path_join(), but doesn't matter)

Reply:
Yeah, I really dislike `strjoina()`. But in this case, it should be OK.